Subject: [PATCH] Notification: Fix no icon shown for data limit notification

When USB connects and data warning limit are enabled, the icon
for data limit is not shown in status bar after data limit reaches.

The reason is that when data limit notification reaches, the total
number of notifications for package android becomes 4, then will
auto-group, the group summary will re-use the firstly enqueued
notification's channel and importance, USB Charging notification is
the firstly enqueued one whose channel has IMPORTANT_MIN importance,
SystemUI will not show the icon for this unimportant notification,
it's considered as ambient notification in SystemUI.

For notifications in package android, they have different icons and
importance, so will not auto-group them.

diff --git a/services/core/java/com/android/server/notification/GroupHelper.java b/services/core/java/com/android/server/notification/GroupHelper.java
index ce805aadec7..4068c2190c6 100644
--- a/services/core/java/com/android/server/notification/GroupHelper.java
+++ b/services/core/java/com/android/server/notification/GroupHelper.java
@@ -53,6 +53,7 @@ public void onNotificationPosted(StatusBarNotification sbn, boolean autogroupSum
                 // Not grouped by the app, add to the list of notifications for the app;
                 // send grouping update if app exceeds the autogrouping limit.
                 synchronized (mUngroupedNotifications) {
+                    if ("android".equals(sbn.getPackageName())) return;
                     Map<String, LinkedHashSet<String>> ungroupedNotificationsByUser
                             = mUngroupedNotifications.get(sbn.getUserId());
                     if (ungroupedNotificationsByUser == null) {
